app/jobs/pull_latest_version_data.rb

```diff
@@ -29,11 +29,37 @@ def save(submission, json_data)
       data: json_data['application']
     )
 
+    # performed here to avoid slow transactions as requires API call to the OS API
+    cached_autograntable = autograntable(submission:)
+
+    PriorAuthorityApplication.transaction do
+      update_submission(submission, json_data)
+
+      autogrant(submission) if cached_autograntable
+    end
+  end
+
+  def autograntable(submission:)
+    # performed here to avoid slow transactions as requires API call to the OS API
+    Autograntable.new(submission:).grantable?
+  rescue LocationService::LocationError
+    false
```

Comment:
If the location error is anything other than a LocationService::NotFoundError then that suggests that something has happened that we may need to react to, like an API key having expired. Should we do a Sentry notification for that as part of the handling?